setup.py

```diff
@@ -20,6 +25,201 @@
+                else:
+                    cudart_path = ctypes.util.find_library('cudart')
+                    if cudart_path is not None:
+                        cuda_path = osp.dirname(cudart_path)
+                    else:
+                        cuda_path = None
+                if cuda_path is not None:
+                    CUDA_HOME = osp.dirname(cuda_path)
+                else:
+                    CUDA_HOME = None
+            cuda_version = find_cuda_version(CUDA_HOME)
+        return cuda_version
+
+    def get_gcc():
+        proc = subprocess.Popen(['gcc', '--version'],
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE)
+        out, err = proc.communicate()
+        gcc_version = re.search(r'[0-9]+\.[0-9]+\.[0-9]+', out.decode())
+        if gcc_version:
+            return gcc_version.group()
+        else:
+            return None
+
+    def _minimal_ext_cmd(cmd):
+        # construct minimal environment
+        env = {}
+        for k in ['SYSTEMROOT', 'PATH', 'HOME']:
+            v = os.environ.get(k)
+            if v is not None:
+                env[k] = v
+        # LANGUAGE is used on win32
+        env['LANGUAGE'] = 'C'
+        env['LANG'] = 'C'
+        env['LC_ALL'] = 'C'
+        out = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, env=env).communicate()[0]
+        return out
+
+    def get_git_hash(fallback='unknown', digits=None):
+        """Get the git hash of the current repo.
+
+        Args:
+            fallback (str, optional): The fallback string when git hash is
+                unavailable. Defaults to 'unknown'.
+            digits (int, optional): kept digits of the hash. Defaults to None,
+                meaning all digits are kept.
+
+        Returns:
+            str: Git commit hash.
+        """
+
+        if digits is not None and not isinstance(digits, int):
+            raise TypeError('digits must be None or an integer')
+
+        try:
+            out = _minimal_ext_cmd(['git', 'rev-parse', 'HEAD'])
+            sha = out.strip().decode('ascii')
+            if digits is not None:
+                sha = sha[:digits]
+        except OSError:
+            sha = fallback
+
+        return sha
+
+    mmcv_root = Path(__file__).parent
+    version_path = mmcv_root / 'mmcv' / 'version.py'
+    gcc_version = get_gcc()
+    cuda_version = get_cuda()
+    git_hash = get_git_hash(digits=7)
+    if EXT_TYPE:
+        torch_version = torch.__version__
+        if '+' not in torch_version:  # eg. 1.8.1 -> 1.8.1+cu111
+            if torch.cuda.is_available():
+                torch_version += '+cu'
+                torch_version += (torch.version.cuda).replace('.', '')
+            else:
+                torch_version += '+cpu'
+    else:
+        torch_version = None
+    with open(version_path, 'r+') as f:
+        lines = f.readlines()
+        last_line = lines[-1]
+        if '__all__' not in last_line:
```

Comment:
It maybe better to put those information after `__version__` rather than `__all__`